viewing_party/party.py

```diff
@@ -1,23 +1,167 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie.update({"title" : title, "genre" : genre, "rating" : rating})
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    print(user_data)
+    watchlist_list_of_dict = user_data.get("watchlist")
+    print(watchlist_list_of_dict)
+
+    for watchlistDict in watchlist_list_of_dict:
+        if watchlistDict.get("title") == title:
+            watchlist_list_of_dict.remove(watchlistDict)
+            user_data["watched"].append(watchlistDict)
+            print(user_data)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    running_total = 0
+    number_of_movies = 0
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    for movie in user_data["watched"]:
+        running_total += movie["rating"]
+        number_of_movies += 1
+    avg = running_total / number_of_movies
+    return avg
 
+def get_most_watched_genre(user_data):
+    #make sure an empty list returns None
+    if len(user_data["watched"]) < 1:
+        most_watched_genre = None 
+        return most_watched_genre
 
+    #establish dictionary there the genre is the key and the value is the intances of viewing ex {"genre": 1}
+    most_watched_genre_dict = {}
+    #loop through watched to add to dictionary
+    for movie in user_data["watched"]:
+        if movie["genre"] not in most_watched_genre_dict:
+            most_watched_genre_dict[movie["genre"]] = 1
+        else: 
+            most_watched_genre_dict[movie["genre"]] += 1
+    #establish variables to hold the count and name of highest genre
+    highest_genre = None
+    highest_count_sofar = 0
+    # loop through most watched dict to update
+    for genre, count in most_watched_genre_dict.items():
+        if count > highest_count_sofar:
+            highest_count_sofar = count
+            highest_genre = genre
+    #this better work

```

Comment:
Solely argument's sake, what's another way we might calculate the highest watched genre with our `most_watched_genre_dict` dictionary and, say, the `max` function? In what ways might `max` help make our function more efficient? Or not?